Tasks/ServiceFabricUpdateManifestsV2/Update-DockerImageSettings.psm1

```diff
@@ -12,18 +12,49 @@ function Update-DockerImageSettings
         $appPackagePathSearchPattern = (Get-VstsInput -Name applicationPackagePath -Require)
         $appPackagePath = Find-VstsFiles -LegacyPattern $appPackagePathSearchPattern -IncludeDirectories
 
-        # Collect the image digest values for the images
+        $imageNamesFilePathSearchPattern = Get-VstsInput -Name imageNamesPath
+        $imageNamesPath = $null
+        if (-not [string]::IsNullOrEmpty($imageNamesFilePathSearchPattern))
+        {
+            $imageNamesPath = Find-VstsFiles -LegacyPattern $imageNamesFilePathSearchPattern
+        }
+
         $imageDigestValues = (Get-Content -LiteralPath $imageDigestsPath).Replace("`r`n", "`n").Split("`n")
+        $imageNames = $null
+        if (($imageNamesPath -ne $null))
+        {
+            $imageNames = (Get-Content -LiteralPath $imageNamesPath).Replace("`r`n", "`n").Split("`n")
+            if ($imageNames.Count -ne $imageDigestValues.Count)
+            {
+                throw (Get-VstsLocString -Key ImageDigestMismatch -ArgumentList @($imageNames.Count, $imageDigestValues.Count))
+            }
+        }
+
+        # Collect the image digest values for the images
         $imageNameToDigestMapping = @{}
-        foreach ($imageDigestValue in $imageDigestValues)
+        for ($ind = 0; $ind -lt $imageDigestValues.Count; $ind++)
         {
-            $slashIndex = $imageDigestValue.IndexOf("/")
-            $hashSeparatorIndex = $imageDigestValue.IndexOf("@")
-            if ($slashIndex -lt 0 -or $hashSeparatorIndex -lt 0)
+            $imageDigestValue = $imageDigestValues[$ind]
+            $imageName = $null
+            if ($imageNames -ne $null)
             {
-                throw (Get-VstsLocString -Key InvalidImageDigestValue -ArgumentList @($imageDigestValue, $imageDigestsPath))
+                $imageName = $imageNames[$ind]
```

Comment:
we should also verify that $imageName's name/repo part is same as $imageDigestValue's. Otherwise most probably this replacement will fail.

Reply:
It doesn't have to be, this task is meant to take care of changing from a local name to the image as pushed to the remote registry. Adding the registry to the beginning is an intended behavior.

Reply:
I meant just the repo part (the '/' in 'name/repo' indicated 'or')

Reply:
This will ensure that repo is matching for both local and remote image.